test/wsdl-load-from-memory-test.js

```diff
@@ -35,10 +56,44 @@ describe('wsdl-load-from-memory-tests', function() {
 
       // Load the wsdl fully once its been created in memory
       stockQuoteWsdl.load(function () {
-        assert.equal(stockQuoteWsdl.definitions['$name'], "StockQuote")
-        done()
+        assert.equal(stockQuoteWsdl.definitions['$name'], "StockQuote");
+        done();
       })
     });
+
+    it('should load a wsdl synchronously with no imports directly from memory or should honour synchronous loading of a wsdl', function (done) {
+
+      var options = {
+        WSDL_CACHE: {}
+      };
+      // Create the initial wsdl directly
+      var stockQuoteWsdl = new WSDL(stockQuoteWsdlContents, undefined, options);
+
+      // Load the wsdl fully once its been created in memory in a sync manner
+      var checkwsdl=stockQuoteWsdl.loadSync();
+      assert.equal(checkwsdl.definitions['$name'], "StockQuote");
+      done();
+    });
+
+    it('should fail to load a wsdl synchronously as wsdl not present in memory', function (done) {
```

Comment:
Hmm how is this test different than the one above (https://github.com/strongloop/strong-soap/pull/186/files#diff-c8a26828475781446eb4ac3eb1ede908R64)? I see that we load both `stockQuoteWsdlContents` and `mainWsdlContents` in the `beforeEach()` code block before executing each test and use the same `new WSDL()` call for both `mainWsdl` and `stockQuoteWsdl`.

Reply:
the loadSync call needs to have all the wsdls in memory cache and so in this test mainWsdl needs to try to access the filessystem (or http) to load     <wsdl:import location="sub.wsdl" namespace="http://example.com/"> and so fails

but the stockquote.wsdl has no imports and so can be loaded in a sync manner and so passes